packages/oidf-client/__tests__/shared/oidfClientAgentLogic.ts

```diff
@@ -0,0 +1,82 @@
+import 'cross-fetch/polyfill'
+import {TAgent} from '@veramo/core'
+import {IOIDFClient} from "../../src";
+import {mockResponses} from "./TrustChainMockResponses";
+import {IJwtService} from "@sphereon/ssi-sdk-ext.jwt-service";
+import nock = require('nock');
+
+type ConfiguredAgent = TAgent<IOIDFClient & IJwtService>
+
+function nockSetup() {
+  const url = new URL(mockResponses[0][0])
+  const url5 = new URL(mockResponses[5][0])
+  nock(url.origin) //https://oidc.registry.servizicie.interno.gov.it
+      .persist()
+      .get(url.pathname) //.well-known/openid-federation
+      .reply(200, mockResponses[0][1], { 'Content-Type': "application/entity-statement+jwt" })
+      .get(url5.pathname).query({ sub: url5.searchParams.get('sub') }) //fetch?sub=https://spid.wbss.it/Spid/oidc/sa
+      .reply(200, mockResponses[5][1], { 'Content-Type': "application/entity-statement+jwt" })
+
+  const url1 = new URL(mockResponses[1][0])
+  const url2 = new URL(mockResponses[2][0])
+  const url3 = new URL(mockResponses[3][0])
+  const url4 = new URL(mockResponses[4][0])
+  nock(url1.origin) //https://spid.wbss.it
+      .persist()
+      .get(url1.pathname) //Spid/oidc/sa/.well-known/openid-federation
+      .reply(200, mockResponses[1][1], { 'Content-Type': "application/entity-statement+jwt" })
+      .get(url2.pathname) //Spid/oidc/rp/ipasv_lt/.well-known/openid-federation
+      .reply(200, mockResponses[2][1], { 'Content-Type': "application/entity-statement+jwt" })
+      .get(url3.pathname).query({ sub: url3.searchParams.get('sub') }) //Spid/oidc/sa/fetch?sub=https://spid.wbss.it/Spid/oidc/rp/ipasv_lt
+      .reply(200, mockResponses[3][1], { 'Content-Type': "application/entity-statement+jwt" })
+      .get(url4.pathname) //Spid/oidc/sa/fetch
+      .reply(200, mockResponses[4][1], { 'Content-Type': "application/entity-statement+jwt" })
+}
+
+export default (testContext: { getAgent: () => ConfiguredAgent; setup: () => Promise<boolean>; tearDown: () => Promise<boolean> }): void => {
+  describe('Credential Store Agent Plugin', (): void => {
+    let agent: ConfiguredAgent
+
+    beforeAll(async (): Promise<void> => {
+      await testContext.setup()
+      agent = testContext.getAgent()
+    })
+
+    afterAll(testContext.tearDown)
+
+    it('should build a trust chain 1', async () => {
+      nockSetup()
+      const trustChain = await agent.resolveTrustChain({
+        entityIdentifier: "https://spid.wbss.it/Spid/oidc/rp/ipasv_lt",
+        trustAnchors:["https://oidc.registry.servizicie.interno.gov.it"]
+      })
+
+      expect(trustChain).toBeDefined()
```

Comment:
this is going to fail if the result is null, as null is a defined value and the result of agent.resolveTrustChain can be null. 

Not sure why the result of agent.resolveTrustChain is so complex anyways

Reply:
Fixed the non-null expectation of the tests. 

I cannot answer the second question, I'm just using the logic that is already there. It's not up to me to decide.